service/src/main/kotlin/fi/espoo/evaka/attendance/StaffAttendanceQueries.kt

```diff
@@ -406,6 +407,21 @@ WHERE employee_id = :employeeId AND tstzrange(start_time, end_time) && tstzrange
         .mapTo<PlannedStaffAttendance>()
         .toList()
 
+fun Database.Read.getPlannedStaffAttendanceForDays(
```

Comment:
It seems like this function should actually return `Map<EmployeeId, List<PlannedStaffAttendance>>`, since we're using the `PlannedStaffAttendanceWithEmployeeId` type just to filter multiple rows by employeeId.

Something like this might work, and you can then remove `PlannedStaffAttendanceWithEmployeeId` and just use map accesses instead of manual filtering:

```kotlin
.map { row -> Pair(row.mapColumn<EmployeeId>("employee_id"), row.mapRow<PlannedStaffAttendance>()) }
.groupBy({ it.first }, { it.second })
```

JDBI also has reduce functions for this kind of stuff, but IMHO they are a bit clunky and this is more readable